bindings/wasm/src/account/storage/traits.rs

```diff
@@ -0,0 +1,307 @@
+    let promise: Promise = Promise::resolve(&self.set_state(did.clone().into(), state.clone().into()));
+    let result: JsValueResult = JsFuture::from(promise).await.into();
+    result.into()
+  }
+
+  /// Removes the keys and any state for the identity specified by `did`.
+  async fn purge(&self, did: &IotaDID) -> AccountResult<()> {
+    let promise: Promise = Promise::resolve(&self.purge(did.clone().into()));
+    let result: JsValueResult = JsFuture::from(promise).await.into();
+    result.into()
+  }
+}
+
+#[wasm_bindgen(typescript_custom_section)]
+const STORAGE: &'static str = r#"
+/** All methods an object must implement to be used as an account storage. */
+interface Storage {
+  /** Sets the account password.*/
+  setPassword: (encryptionKey: Uint8Array) => Promise<void>;
+
+  /** Write any unsaved changes to disk.*/
+  flushChanges: () => Promise<void>;
+
+  /** 
+   * Attempt to obtain the exclusive permission to modify the given `did`.
+   * The caller is expected to make no more modifications after the lease has been dropped.
+   * Returns an error if already leased.
+  */
+  leaseDid: (did: DID) => Promise<DIDLease>;
+
+  /** Creates a new keypair at the specified `location`.*/
+  keyNew: (did: DID, keyLocation: KeyLocation) => Promise<string>;
+
+  /** Inserts a private key at the specified `location`.*/
+  keyInsert: (did: DID, keyLocation: KeyLocation, privateKey: string) => Promise<string>;
+
+  /** Returns `true` if a keypair exists at the specified `location`.*/
+  keyExists: (did: DID, keyLocation: KeyLocation) => Promise<boolean>;
+
+  /** Retrieves the public key at the specified `location`.*/
+  keyGet: (did: DID, keyLocation: KeyLocation) => Promise<string>;
+
+  /** Deletes the keypair specified by `location`.*/
+  keyDel: (did: DID, keyLocation: KeyLocation) => Promise<void>;
+
+  /** Signs `data` with the private key at the specified `location`.*/
+  keySign: (did: DID, keyLocation: KeyLocation, data: Uint8Array) => Promise<Signature>;
+
+  /** Returns the chain state of the identity specified by `did`.*/
+  chainState: (did: DID) => Promise<ChainState>;
+
+  /** Set the chain state of the identity specified by `did`.*/
+  setChainState: (did: DID, chainState: ChainState) => Promise<void>;
+
+  /** Returns the state of the identity specified by `did`.*/
+  state: (did: DID) => Promise<IdentityState>;
+
+  /** Sets a new state for the identity specified by `did`.*/
+  setState: (did: DID, identityState: IdentityState) => Promise<void>;
+
+  /** Removes the keys and any state for the identity specified by `did`.*/
+  purge: (did: DID) => Promise<void>;
+
+  /** Returns the last generation that has been published to the tangle for the given `did`.*/
+  publishedGeneration: (did: DID) => Promise<Generation>;
+
+  /** Sets the last generation that has been published to the tangle for the given `did`.*/
+  setPublishedGeneration: (did: DID, generation: Generation) => Promise<void>;
```

Comment:
I assume this will be removed in the napi or epic branch?

Reply:
Yes. Handled on #660